include/Basis/SpinorBasis/USpinorBasis.hpp

```diff
@@ -0,0 +1,359 @@
+    }
+
+
+    /**
+     *  Rotate the spinor basis to another one using the given unitary transformation matrix
+     * 
+     *  @param U            the unitary transformation matrix that transforms both the alpha- and beta components
+     */
+    void rotate(const TransformationMatrix<ExpansionScalar>& U) {
+        this->rotate(SpinComponent::ALPHA);
+        this->rotate(SpinComponent::BETA);
+    }
+
+    /**
+     *  Rotate the spinor basis of the requested component to another one using the unitary transformation matrix that corresponds to the given Jacobi rotation parameters
+     * 
+     *  @param jacobi_rotation_parameters       the Jacobi rotation parameters (p, q, angle) that are used to specify a Jacobi rotation: we use the (cos, sin, -sin, cos) definition for the Jacobi rotation matrix
+     * 
+     *  @note this function is only available for real spinor bases because Jacobi rotation parameters generate real rotations
+     */
+    template<typename Z = ExpansionScalar>
+    enable_if_t<std::is_same<Z, double>::value> rotate(const JacobiRotationParameters& jacobi_rotation_parameters, const SpinComponent& component) {
+        this->spinor_basis[component].rotate(jacobi_rotation_parameters);
+    }
+
+    /**
+     *  Rotate the spinor basis for both the alpha- and beta components to another one using the unitary transformation matrix that corresponds to the given Jacobi rotation parameters
+     * 
+     *  @param jacobi_rotation_parameters       the Jacobi rotation parameters (p, q, angle) that are used to specify a Jacobi rotation: we use the (cos, sin, -sin, cos) definition for the Jacobi rotation matrix
+     * 
+     *  @note this function is only available for real spinor bases because Jacobi rotation parameters generate real rotations
+     */
+    template<typename Z = ExpansionScalar>
+    enable_if_t<std::is_same<Z, double>::value> rotate(const JacobiRotationParameters& jacobi_rotation_parameters) {
+        this->rotate(jacobi_rotation_parameters, SpinComponent::ALPHA);
+        this->rotate(jacobi_rotation_parameters, SpinComponent::BETA);
+    }
+
+    /**
+     *  Transform the spinor basis another one using the given transformation matrix
+     * 
+     *  @param component                the spin component
+     * 
+     *  @param T            the transformation matrix that transforms the requested component
+     */
+    void transform(const TransformationMatrix<ExpansionScalar>& T, const SpinComponent& component) {
+         this->spinor_basis[component].transform(T);
+    }
+
+    /**
+     *  Transform the spinor basis another one using the given transformation matrix
+     * 
+     *  @param T            the transformation matrix that transforms both the alpha- and beta components
+     */
+    void transform(const TransformationMatrix<ExpansionScalar>& T) {
+         this->transform(T, SpinComponent::ALPHA);
+         this->transform(T, SpinComponent::BETA);
+    }
```

Comment:
Should we include some safeguard for the given dimensions of the transformation matrix?

Reply:
At the level of the transformation matrix?

Reply:
I would say at this level: what should happen if we supply a `TransformationMatrix` of dimension `MxM`, i.e. the dimension of the 'full' transformation matrix?

Reply:
Eigen will throw an assertion error, I guess it should be in the comments though? And I wouldn't consider accepting MxM. Because then we also check whether or not it has two-blocks for K_a x K_a and K_b x K_b.